src/WebDAVAdapter.php

```diff
@@ -275,7 +275,14 @@ public function getMimetype($path)
      */
     protected function normalizeObject(array $object, $path)
     {
-        if (! isset($object['{DAV:}getcontentlength']) or $object['{DAV:}getcontentlength'] == "") {
+        $hasNotContentLength = !isset($object['{DAV:}getcontentlength']) ||
+            $object['{DAV:}getcontentlength'] == "";
+
+        $isDir = $hasNotContentLength || isset($object['{DAV:}resourcetype']) &&
+            $object['{DAV:}resourcetype'] instanceof ResourceType &&
+            $object['{DAV:}resourcetype']->is('{DAV:}collection');
```

Comment:
Please use parenthesis around the or condition or around the and conditions to highlight the correct meaning of the condition.

Otherwise it's prone to errors and regressions in the future, if something is changed or maybe reordered.